custom/enikshay/management/commands/drtb_2_to_2_dot_1_migration.py

```diff
@@ -0,0 +1,287 @@
+            props_to_update['close_reason'] = episode_case_props.get('treatment_outcome')
+            props_to_update['treatment_outcome'] = ""
+
+        # Set episode.case_definition = microbiological if episode_type = confirmed_drtb
+        if episode_case_props.get('episode_type') == "confirmed_drtb":
+            props_to_update['case_definition'] = 'microbiological'
+
+        # episode.dstb_to_drtb_transition_no_initiation = yes
+        # IF episode_type = confirmed_tb AND (treatment_initiated = 'no' or treatment_initiated = '')
+        # AND there exists another episode with episode_type=confirmed_drtb for the same occurrence
+        if (episode_case_props.get('episode_type') == "confirmed_tb" and
+                episode_case_props.get('treatment_initiated', "") in ["no", ""] and
+                any_other_confirmed_drtb_episode(episode_case, episode_cases)):
+            props_to_update['dstb_to_drtb_transition_no_initiation'] = 'yes'
+        return props_to_update
+
+    @staticmethod
+    def test_case_updates(test_case, recently_opened_episode_case, occurrence_case):
+        def get_recently_opened_episode_case(occ_case):
+            accessor = CaseAccessors(DOMAIN)
+            episode_cases = [case for case in accessor.get_reverse_indexed_cases([occ_case.get_id])
+                             if case.type == CASE_TYPE_EPISODE and
+                             case.dynamic_case_properties().get('episode_type') == "confirmed_tb"]
+            episode_cases.sort(key=lambda x: x.opened_on, reverse=True)
+            if episode_cases:
+                return episode_cases[0]
+
+        props_to_update = {}
+        test_case_props = test_case.dynamic_case_properties()
+        if (test_case_props.get('rft_general') == "follow_up_dstb" and
+                test_case_props.get('result_recorded') == "yes"):
+            if not recently_opened_episode_case:
+                recently_opened_episode_case = get_recently_opened_episode_case(occurrence_case)
+            if recently_opened_episode_case:
+                props_to_update['debug'] = {'episode_case_id': recently_opened_episode_case.get_id}
+                treatment_initiation_date = (recently_opened_episode_case.dynamic_case_properties()
+                                             .get('treatment_initiation_date'))
+                props_to_update['debug']['treatment_initiation_date'] = treatment_initiation_date
+                date_tested = test_case_props.get('date_tested')
+                props_to_update['debug']['date_tested'] = date_tested
+                if treatment_initiation_date and date_tested:
+                    d1 = datetime.strptime(date_tested, "%Y-%m-%d")
+                    d2 = datetime.strptime(treatment_initiation_date, "%Y-%m-%d")
+                    props_to_update['rft_dstb_follow_up_treatment_month'] = (
+                        (d1.year - d2.year) * 12 + d1.month - d2.month)
+        return recently_opened_episode_case, props_to_update
```

Comment:
Looks like `recently_opened_episode_case` is potentially set in this method, which I assume is why it's being returned, but it's being thrown away (the update doesn't make it back to the `handle` method).

Reply:
oh i see.. Ya need to pass that back up. Updating.

Reply:
Updated [here](https://github.com/dimagi/commcare-hq/pull/17940/commits/a6900c84f57cb32c96eeee2d8816708a086cadce)